hal/armv7m/stm32/_init.S

```diff
@@ -130,20 +140,25 @@ _syscall_dispatch:
 	bx lr
 
 _syscallend:
-	pop {lr}
+	/* return value in r0 */
+	ldr lr, [sp, #8] /* psp */
 
-	ldmia sp!, {r1-r8}
-	mov r1, r0 /* Put result to user's r0 */
-	stmdb lr!, {r1-r8}
+	add r10, sp, #((12 * 4) + (5 * 4))
 
-	ldmia sp!, {r4-r8}
+	ldm r10, {r1-r3} /* lr, pc, psr */
+	stmdb lr!, {r1-r3}
+
+	add sp, sp, #12
+	ldmia sp!, {r4-r11}
+	add sp, sp, #36
 
 	/* Switch stacks */
-	mov r0, #3
-	msr control, r0
+	mov r1, #3
+	msr control, r1
 	isb
 
-	ldmia sp!, {r0-r3, r12, lr}
+	add sp, sp, #(5 * 4)
+	pop {lr}
```

Comment:
A tiny window where the stack is not aligned, but interrupts are enabled - tiny, but still might cause problems

Reply:
Hmm, from what I see stm32 and imxrt implementation diverged, why? There shouldn't be any difference apart from FPU stuff

Reply:
The implementations were a little bit different originally, and I couldn't find eval kit to test this further